### ISSUE TITLE
Add a SwerveModule class to greatly reduce code repetition in Drivetrain

### ISSUE THREAD

**Issue by brettle** [DeepBlueRobotics/SwerveDrive#8]
In many places in Drivetrain, nearly identical code is repeated 4 times, with the only differences being which swerve module is being referred to. This makes the code more voluminous than necessary, makes it easier to introduces bugs, and harder to analyze, debug, and maintain. Encapsulating the things that are different between modules (e.g. motor controller objects, encoder objects, constants, labels) into a SwerveModule class would allow the various places with repeated code to be replaced with a "for" loop over the SwerveModules.

**Comment by brettle** [comment]
It looks like @alexander-mcdowell started doing this, but there is still a lot of repeated code in `Drivetrain`. If you make the `SwerveModule` constructor take a string label (e.g. "FL", "FR", "BL", "BR"), and some `DriveConstants`, and add a `SwerveModule.updateSmartDashboard()` method, you can greatly simplify `Drivetrain`.

**Comment by brettle** [comment]
This is looking much better, but can still be improved. To force yourself to the finish, replace:
```
  public SwerveModule moduleFL;
  public SwerveModule moduleFR;
  public SwerveModule moduleBL;
  public SwerveModule moduleBR;
```
with
```
public SwerveModule[] modules = new SwerveModule[4];
```
 and make the modifications needed so that you only refer explicitly to `modules[0]`, `modules[1]`, `modules[2]`, and `modules[3]` in the `Drivetrain` constructor. This will force you to keep everything that only an individual module needs to know about encapsulated in `SwerveModule`.

**Comment by brettle** [comment]
Almost there. The following code contains multiple bugs:

```    // Move the modules based on desired (normalized) speed, desired angle, max speed, drive modifier, and whether or not to reverse turning.
    for (int i = 0; i < 4; i++) {
      modules[i].move(moduleStates[0].speedMetersPerSecond, moduleStates[0].angle.getRadians(), 
                  Constants.DriveConstants.maxSpeed,  -Constants.DriveConstants.kDriveModifier, Constants.DriveConstants.reversed[i]);
    }
```
Also, the method `SwerveModule.updateSmartDahsboard()` should probably be renamed to `SwerveModule.updateSmartDashboard()`. :smile:

**Comment by brettle** [comment]
Fixed as of aa379e7cac77686fea476a7120a33b70430b7872.
